packages/notifications/src/styled/StyledAlert.ts

```diff
@@ -25,10 +25,12 @@ const colorStyles = (props: IStyledAlertProps & ThemeProps<DefaultTheme>) => css
 /**
  * Supports all `<div>` props
  */
-export const StyledAlert = styled(StyledBase).attrs({
+export const StyledAlert = styled(StyledBase).attrs(props => ({
   'data-garden-id': COMPONENT_ID,
-  'data-garden-version': PACKAGE_VERSION
-})<IStyledAlertProps>`
+  'data-garden-version': PACKAGE_VERSION,
+  role: 'alert' as string,
+  ...props
```

Comment:
Pls don’t spread props as attrs. This is bad for current or future HTML. Logically apply the role, instead.

Reply:
I thought props got forwarded under the hood by styled-components, so I didn't think spreading props into attrs would hurt. I thought only HTML attributes get applied. 

Could you give an example of issues that would arise when props and spread into attrs?

Reply:
> I thought only HTML attributes get applied

This is true by default, but there are instances where it doesn't work as expected https://github.com/styled-components/styled-components/issues/439

This could also influence type-inference with TypeScript depending on custom props.

It may just be safer to use `role: (props.role || 'alert') as string`

Reply:
I see, thanks for the clarification and suggestion @austin94. I'll go ahead and use `role: (props.role || 'alert') as string`.

Reply:
I just realized my original suggestion might not work 100% of the time. Does this work with truthy/falsy? We may want to use `props.role === undefined ? 'alert' : props.role`

Reply:
Yeah, @austin94 that's the treatment.

@hzhu in the current code, we see `type` bleeding through as an attribute to the HTML. Austin's code will fix that.

In addition please add documentation and testing per my comments above.